### DISCUSSION TITLE
Put Models Where?

### DISCUSSION THREAD

**Discussion by y0himba** [lllyasviel/Fooocus#192] [Q&A]
Hello.

I have a mix of `.safetensors`. `.ckpt`, and `.pth` models that I would like to use in Fooocus.  I know where to place the `.safetensors`, what about the rest?  Can I place them in `/Fooocus/models/checkpoints/` as well?

To be honest I'm not even sure if Fooocus will use custom models, but I assume it will...

**Comment by SergioBoySV** [comment]
Hey, I know it's been a while since you posted this, but I am wondering where to put my own models as well . Would love to know if you figured it out!

**Comment by y0himba** [comment]
Unfortunately no, and there has been no answer in all this time.

**Reply by SergioBoySV** [reply]
Dang that's alright. 

I'm guessing here because I'm still learning, but perhaps you're meant to put the `.ckpt` file in `/Fooocus/models/checkpoints` and insert the model name into the `put_checkpoints_here` file in the same directory. 

I am going to give it a shot.

**Reply by XIUMAII** [reply]
Hi! If you did not find it for .safetensors it's in \Fooocus\models\loras for LoRas models or  Fooocus\models\checkpoints depend of what kind of models ofr models XL like and other. This is the little I can know

**Reply by SergioBoySV** [reply]
Interesting. Unfortunately my attempts didn't work. 

First I edited the `config.txt` file and added this line: 
`"default_model": "my_model.ckpt"`
As well as added the string "my_model" to `Fooocus/models/checkpoints/put_checkpoints_here`

Second, I tried changing the base model inside the UI itself. This did not work either
<img width="501" alt="Screenshot 2023-12-01 at 1 36 22 PM" src="https://github.com/lllyasviel/Fooocus/assets/85202224/3d608727-47eb-487a-a948-728d45d53944">

I am a beginner so these are just some of my attempts. Anyways, thanks for the response @XIUMAII!

**Reply by botzero-net** [reply]
You should just be able to drop the ckpt or safetensors file in the checkpoints folder.  You don't have to edit the txt file in that folder (it's just there as a FYI) or edit the config file--the new models will show up under Base Model drop down you have in your screen shot.

This is how it works reliably for me, though only with XL models.

**Reply by SergioBoySV** [reply]
Ah okay that's interesting @jonathanmiser9. When I changed the base model from the dropdown menu, I got an error: 
`NotImplementedError: You have selected base model other than SDXL. This is not supported yet.`

I'm not familiar with what an XL model is, but I'm willing to bet that the error lies with my `.ckpt` file not being an XL model. I will do some research. Thank you for your response. At least it's good to know that Fooocus supports other models!

**Reply by botzero-net** [reply]
DynavisionXL is pretty good and available on Civit!
